### DISCUSSION TITLE
Ability to show log book view next to an automation trigger or condition

### DISCUSSION THREAD

**Discussion by niemyjski** [home-assistant/frontend#7811] [Automations / scripts / scenes]
I find myself using the logbook in a separate tab to find out why an automation didn't trigger or match a condition. It would be super nice if I could click a button or view inline what happened to that device or entity during a time frame.

**Comment by KTibow** [comment]
Maybe this could be added to the quick bar?

**Reply by niemyjski** [reply]
It could and would be super helpful. But I think the dead whitespace to the left hand side could maybe show the last 3 events of the entity you are trying to use or maybe a button that shows in a modal so you can quickly see different states. This is helpful if an automation doesn't trigger and you think it should have so you go and inspect the automation and then you look at the last three states of each condition or trigger and you could see why something didn't work.

**Reply by KTibow** [reply]
Could you try to make a simple mockup of where it could be?

**Reply by niemyjski** [reply]
I'm not a designer by any means but I think there are many different ways to implement this in a useful way.

1. We could either put a narrow history in the footer of each trigger or condition. I think if you click on this it would bring up a modal with the log view.

2. We could add a dropdown option on the trigger to show the log book using the existing dropdown.

3. We could put a log view next to the entity on the left, this is less ideal as it makes the ui pretty busy
![image](https://user-images.githubusercontent.com/1020579/100782433-07af2680-33d2-11eb-979d-3b9d16161f57.png)

**Reply by niemyjski** [reply]
I just want to see a quick history so I can visually see if something changed and should have been triggered but wasn't. I'm a visual person. I think you may want to compose multiple histories in some circumstances like maybe for delayed or wait until actions (to see the current state).

**Reply by KTibow** [reply]
Okay. Now we just need to add this to the code. I would do this, but I'm not familiar with handling history. It could be you could just embed a Lovelace card that does that.
As for your second example: Think about mobile. Would the logbook be below or above or inside?

**Reply by niemyjski** [reply]
I think embedding a Lovelace card would work but some have different layouts for history card? Would it be better to see a log book of all trigger entities, a log book of all conditions etc...?

For mobile maybe they are hidden or it pops up a modal from the options menu on the trigger or condition?

**Reply by KTibow** [reply]
How about [this icon](https://mdi.bessarabov.com/icon/history) in the device/entity text fields?
